File: backend/core_app/domain/entities/class.go
package entities

import (
	"time"
)

type Class struct {
	BaseEntity
	ClassName           string `gorm:"size:255" json:"class_name"`
	ClassCode           string `gorm:"size:10;unique" json:"class_code"`
	ClassDescription    string `gorm:"type:text" json:"class_description"`
	ClassManagerZCodeID uint64 `json:"class_manager_zcode_id"`
	ClassManagerName    string `json:"class_manager_name"`
}

type Lecture struct {
	BaseEntity
	LectureName        string     `gorm:"size:255" json:"lecture_name"`
	LectureDescription string     `gorm:"type:text" json:"lecture_description"`
	ClassID            uint       `json:"class_id"`
	StartTime          *time.Time `json:"start_time"`
	EndTime            *time.Time `json:"end_time"`
	LecturerZCodeID    uint64     `json:"lecturer_zcode_id"`
	LecturerName       string     `gorm:"size 255" json:"lecturer_name"`
}

type ClassParticipants struct {
	BaseEntity
	ClassID     uint   `json:"class_id"`
	ClassName   string `json:"class_name"`
	UserZCodeID uint64 `json:"user_zcode_id"`
	Username    string `json:"username"`
	UserRole    string `json:"user_role"`
}

File: frontend/src/dto/response/class.ts
// src/dto/class.ts

export interface ClassInfo {
    class_id: number;
    created_at: string | null;
    class_name: string;
    class_code: string;
    class_description: string;
    class_manager_zcode_id: bigint;
    class_manager_name: string;
}


export interface LectureInfo {
    lecture_id: number;
    created_at: string | null;
    lecture_name: string;
    lecture_description: string;
    class_id: number;
    start_time: string | null;
    end_time: string | null;
    lecturer_z_code_id: string;
    lecturer_name: string;
}

export interface ClassParticipantInfo {
    class_participant_id: number;
    class_id: number;
    class_name: string;
    user_zcode_id: bigint;
    username: string;
    user_role: string;
}

export interface UserJoinedClassesInfo {
    class_participant_id:number;
    class_id:number;
    class_name:string;
    class_code:string;
    class_manager_name:string;
    created_at:string;
}